kernel/src/scan/file_stream.rs

```diff
@@ -142,9 +173,91 @@ impl LogReplayScanner {
             })
             .collect()
     }
+
+    fn get_add_transform_expr(&self) -> Expression {
+        Expression::Struct(vec![
+            Expression::column("add.path"),
+            Expression::column("add.size"),
+            Expression::column("add.modificationTime"),
+            Expression::column("add.deletionVector"),
+            Expression::Struct(vec![Expression::column("add.partitionValues")]),
+        ])
+    }
+
+    fn process_scan_batch(
+        &mut self,
+        expression_handler: &dyn ExpressionHandler,
+        actions: &dyn EngineData,
+        is_log_batch: bool,
+    ) -> DeltaResult<(Box<dyn EngineData>, Vec<bool>)> {
+        // apply data skipping to get back a selection vector for actions that passed skipping
+        // note: None implies all files passed data skipping.
+        let filter_vector = self
```

Comment:
there is common code in here with `process_batch` that i'll consolidate once we settle on the final approach